### DISCUSSION TITLE
Support docstring templates / generate docstrings

### DISCUSSION THREAD

**Discussion by assafmendelson** [microsoft/pylance-release#5287] [Enhancement]
Provide auto fill of docstring.
The idea would be that if I type """ right at a docstring position (file, class method), it would generate a proper template for it.
A simple example would be to start typing """ right after def f(a): and it would immediatly expand it to:

def f(a):
    """
    :param a
    :return
    """

**Comment by zooba** [comment]
Yeah, I like it. This is especially handy if microsoft/PTVS#758 is done too.

**Comment by RussBaz** [comment]
I'm sceptical of this feature if it is always on.

I do not generally like IDE writing a piece of code on its own when I am not asking it to do so explicitly (even if it is just a docstring).

Do you think it will be better, perhaps, if it could be triggered by a tab button or if it could have a switch in settings to prevent it from running automatically when so desired?

**Comment by den-run-ai** [comment]
can this be considered under code snippets?

**Comment by int19h** [comment]
I don't think VS code snippets are sufficiently context-aware to pull off something like this.

What this reminds me of actually is GhostDoc: https://www.youtube.com/watch?t=21&v=xeZqoCGcHSU

**Comment by den-run-ai** [comment]
for now sublime can do this, works perfectly! :)

https://packagecontrol.io/packages/AutoDocstring

**Comment by alexranaldi** [comment]
I like the idea, but we should have the ability to disable it too.

**Comment by sonictk** [comment]
Upvote +1, I would definitely love this in PTVS.

**Comment by lmazuel** [comment]
Upvote :)

**Comment by gramster** [comment]
I'm going to move this issue to vscode-python; it has more chance of getting traction there first.

**Comment by luabud** [comment]
Just FYI for those who are interested, there's a 3rd party extension that provides that: https://marketplace.visualstudio.com/items?itemName=njpwerner.autodocstring

**Comment by debonte** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.
